Code/Editor/MainWindow.cpp

```diff
@@ -514,6 +492,16 @@ void MainWindow::Initialize()
     ActionOverrideRequestBus::Event(
         GetEntityContextId(), &ActionOverrideRequests::SetupActionOverrideHandler, this);
 
+    if (auto imGuiManager = AZ::Interface<ImGui::IImGuiManager>::Get())
+    {
+        auto handleImGuiStateChangeFn = [](bool enabled)
+        {
+            EditorWindowUIRequestBus::Broadcast(&EditorWindowUIRequests::SetEditorUiEnabled, enabled);
+        };
+        m_handleImGuiStateChangeHandler = ImGui::IImGuiManager::ImGuiSetEnabledEvent::Handler(handleImGuiStateChangeFn);
+        imGuiManager->ConnectImGuiSetEnabledChangedHander(m_handleImGuiStateChangeHandler);
```

Comment:
This can be simplified by following a pattern that the SettingsRegistryImpl.cpp takes where it accepts a functor and registers a handler for the user.
See the `RegisterNotifier` function in SettingsRegistryImpl.cpp

```suggestion
        auto handleImGuiStateChangeFn = [](bool enabled)
        {
            EditorWindowUIRequestBus::Broadcast(&EditorWindowUIRequests::SetEditorUiEnabled, enabled);
        };
        m_handleImGuiStateChangeHandler = imGuiManager->ConnectImGuiSetEnabledChangedHandler(AZStd::move(handleImGuiStateChangeFn));
```

Reply:
I tried this code but it had the same problem I had before adding the member variable. The code didn't run when called.